base/abstractdict.jl

```diff
@@ -513,6 +513,18 @@ end
 getindex(t::AbstractDict, k1, k2, ks...) = getindex(t, tuple(k1,k2,ks...))
 setindex!(t::AbstractDict, v, k1, k2, ks...) = setindex!(t, v, tuple(k1,k2,ks...))
 
+get!(t::AbstractDict, key, default) = get!(() -> default, t, key)
+function get!(default::Callable, t::AbstractDict{K,V}, key0) where K where V
+    key = convert(K, key0)
+    if !isequal(key, key0)
```

Comment:
Ah, this is probably why we didn't have this generic definition before --- without knowing more about the type of dictionary, it's not clear which keys are valid. Maybe we should not do the check, and leave it to `setindex!`? Of course, that calls the `default` function even if the value won't be used, but if `setindex!` is going to throw an error I'm not sure it matters.